### PR TITLE
feat(wallet-ui): add terms dialog on first pageview

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/agoric-sdk/issues/6491

In draft until we get final confirmation for content/behavior

image

OLD screenshot:

image

### CODE REVIEW

File: packages/wallet/ui/src/components/AppBar.jsx [hunk 1]
```diff
@@ -63,13 +66,21 @@ const useStyles = makeStyles(theme => ({
...
 // Exported for testing only.
-export const AppBarWithoutContext = ({ connectionComponent }) => {
+export const AppBarWithoutContext = () => {
   const theme = useTheme();
   const classes = useStyles(theme);
 
-  const [dialogOpen, setDialogOpen] = useState(false);
-  const handleClosed = () => {
-    setDialogOpen(false);
+  const [areTermsAccepted, setAreTermsAccepted] = useState(
```
Comment (turadg) [inline packages/wallet/ui/src/components/AppBar.jsx:73]: for future, consider sticking to `is` convention. e.g. `isDisclaimerAccepted`.

File: packages/wallet/ui/src/components/AppBar.jsx [hunk 2]
```diff
@@ -118,7 +138,6 @@ export const AppBarWithoutContext = ({ connectionComponent }) => {
 };
 
 export default withApplicationContext(AppBarWithoutContext, context => ({
-  connectionComponent: context.connectionComponent,
```
Comment (turadg) [inline packages/wallet/ui/src/components/AppBar.jsx:121]: 👍

File: packages/wallet/ui/src/store/Terms.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { TERMS_KEY, maybeSave, maybeLoad } from '../util/storage.js';
+
+// The index of the current revision of the terms agreement. This should
+// increment every time the terms change.
+export const LATEST_TERMS_INDEX = 0;
+
+export const agreeToTerms = (/** @type {number} */ index) =>
+  maybeSave(TERMS_KEY, index);
+
+export const getTermsAcceptance = () => maybeLoad(TERMS_KEY) ?? -1;
```
Comment (turadg) [inline packages/wallet/ui/src/store/Terms.js]:
```suggestion
export const getTermsIndexAccepted = () => maybeLoad(TERMS_KEY) ?? -1;
```

File: packages/wallet/ui/src/store/Terms.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { TERMS_KEY, maybeSave, maybeLoad } from '../util/storage.js';
+
+// The index of the current revision of the terms agreement. This should
+// increment every time the terms change.
+export const LATEST_TERMS_INDEX = 0;
```
Comment (turadg) [inline packages/wallet/ui/src/store/Terms.js:5]: I wish we had unique URLs for each set of terms but not a blocker

**Review by turadg — APPROVED** [review]

File: packages/wallet/ui/src/components/AppBar.jsx (hunk 1 above)
Comment (samsiegart) [reply packages/wallet/ui/src/components/AppBar.jsx:73]: In the style guide I learned from, both 'is' and 'are' are okay depending on if it's plural, also 'can', 'should', 'has', etc. I forget the grammatical term for this type of phrase though.

File: packages/wallet/ui/src/store/Terms.js (hunk 3 above)
Comment (samsiegart) [reply packages/wallet/ui/src/store/Terms.js]: Done

**Pull request merged**